internal/controller/resilient_deployment_workflow.go

```diff
@@ -0,0 +1,253 @@
+
+	return nil
+}
+
+func NewResilienceRateLimitingReconciler() *ResilienceRateLimitingReconciler {
+	return &ResilienceRateLimitingReconciler{}
+}
+
+type ResilienceRateLimitingReconciler struct{}
+
+func (r *ResilienceRateLimitingReconciler) Subscription() controller.Subscription {
+	return controller.Subscription{
+		ReconcileFunc: r.reconcile,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &kuadrantv1beta1.KuadrantGroupKind},
+		},
+	}
+}
+
+func (r *ResilienceRateLimitingReconciler) reconcile(ctx context.Context, _ []controller.ResourceEvent, _ *machinery.Topology, _ error, state *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("ResilienceRateLimitingReconciler")
+
+	logger.V(level).Info("ResilienceRateLimitingReconciler Task", "status", "started")
+	defer logger.V(level).Info("ResilienceRateLimitingReconciler Task", "status", "completed")
+	if !experimentalFeatureEnabledSate(state) {
+		logger.V(level).Info("Experimental resilience feature is not enabled, early exit", "status", "exiting")
+		return nil
+	}
+	logger.V(level).Info("Experimental resilience feature is enabled", "status", "processing")
+
+	return nil
+}
+
+func NewResilienceCounterStorageReconciler(client *dynamic.DynamicClient) *ResilienceCounterStorageReconciler {
+	return &ResilienceCounterStorageReconciler{
+		Client: client,
+	}
+}
+
+type ResilienceCounterStorageReconciler struct {
+	Client *dynamic.DynamicClient
+}
+
+func (r *ResilienceCounterStorageReconciler) Subscription() controller.Subscription {
+	return controller.Subscription{
+		ReconcileFunc: r.reconcile,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &kuadrantv1beta1.KuadrantGroupKind},
+			{Kind: &kuadrantv1beta1.LimitadorGroupKind},
+		},
+	}
+}
+
+func (r *ResilienceCounterStorageReconciler) reconcile(ctx context.Context, _ []controller.ResourceEvent, topology *machinery.Topology, _ error, state *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("ResilienceCounterStorageReconciler")
+
+	logger.V(level).Info("ResilienceCounterStorageReconciler Task", "status", "started")
+	defer logger.V(level).Info("ResilienceCounterStorageReconciler Task", "status", "completed")
+	if !experimentalFeatureEnabledSate(state) {
+		logger.V(level).Info("Experimental resilience feature is not enabled, early exit", "status", "exiting")
+		return nil
+	}
+	logger.V(level).Info("Experimental resilience feature is enabled", "status", "processing")
+
+	kObj := GetKuadrantFromTopology(topology)
+	lObj := GetLimitadorFromTopology(topology)
+	if lObj == nil {
+		logger.V(level).Info("limitador resource has not being created yet.")
+		return nil
+	}
+
+	if !r.isConfigured(kObj) {
+		logger.V(level).Info("CounterStorage not configured", "status", "exiting")
+		return nil
+	}
+	logger.V(level).Info("CounterStorage configured", "status", "contiune")
+
+	lObj.Spec.Storage = kObj.Spec.Resilience.CounterStorage
+	err := r.updateLimitador(ctx, lObj)
```

Comment:
update unconditionally?

Reply:
When it said out loud again, this might not be the best thing. After the last failure of the built in go comparison function to compare the limits for the limitador CR I felt this was the easier option. 

I will go back, and take a second look.